code/game/objects/items/devices/quantum_keycard.dm

```diff
@@ -23,8 +46,15 @@
 	to_chat(user, span_notice("You start pressing [src]'s unlink button..."))
 	if(do_after(user, 40, target = src))
 		to_chat(user, span_notice("The keycard beeps twice and disconnects the quantum link."))
-		qpad = null
+		set_pad()
+
+/obj/item/quantum_keycard/proc/set_pad(obj/machinery/quantumpad/new_pad)
+	qpad = new_pad
+
+	if(!istype(new_pad))
+		set_greyscale(initial(greyscale_colors))
+		name = initial(name)
+		return
 
-/obj/item/quantum_keycard/update_icon_state()
-	icon_state = qpad ? "quantum_keycard_on" : initial(icon_state)
-	return ..()
+	var/new_color = is_type_in_list(get_area(new_pad), gags_coloring, zebra = TRUE) || COLOR_WEBSAFE_DARK_GRAY
```

Comment:
I can't believe this actually fucking works but it actually fucking works

Reply:
the magic of `zebra`